torch/testing/_internal/common_distributed.py

```diff
@@ -878,6 +889,77 @@ def _check_return_codes(self, elapsed_time) -> None:
     def is_master(self) -> bool:
         return self.rank == 0
 
+# Utility base class for distributed Multi Process Test cases
+# This abstracts the PG creation and deletion, the backends are selected based
+# on device type. The tests functions can be instantiated per device type using
+# common_device_type.instantiate_device_type_tests
+# other backends can add entry in backend() function
+class DistributedTestBase(MultiProcessTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self._spawn_processes()
+
+    def tearDown(self):
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    def backend(self, device) -> str:
+        if "cuda" in device:
+            return "nccl"
+        elif "hpu" in device :   # intel gaudi
+            return "hccl"
+        else :
+            return "gloo"
+
+    def device_handle(self, device) :
+        """
+        Get the torch module for the given device type.
+
+        This method is used to handle different device types in distributed testing.
+        It extracts the device name from the input string and returns the corresponding
+        torch module if available.
+
+        Args:
+            device (str): A string representing the device, e.g., "cuda:0", "cpu", "hpu".
+
+        Returns:
+            Union[torch.device, None]: The corresponding torch module if available,
+                                       or None if the device is not found but is an attribute of torch.
+
+        Raises:
+            RuntimeError: If the device library is not found in torch.
+
+        Example:
+            >>> self.device_handle("cuda:0")
```

Comment:
This line raises a CI failure.
I think this function can be refined by https://github.com/pytorch/pytorch/blob/e9fb2c6abe08e6d0d6c830dfe3f38e7917371d65/torch/__init__.py#L2639

Reply:
Fixed in: 04ada419407cd75565afaa7748817c2a4454ecfe